### PR TITLE
Update to LSP 1.7.0

### PR DESCRIPTION
**Pull request by aryx**
test plan:
```
$ semgrep-core -lang ocaml -lsp -e '($E : string ref)' src/cli/Main.ml
src/cli/Main.ml:339
   Common.write_file ~file:tmp (Common.read_file !rules_file)
src/cli/Main.ml:374
   if !use_parsing_cache = "" then f ()
src/cli/Main.ml:402
   let dir = !use_parsing_cache in
...

```

PR checklist:
- [x] documentation is up to date
- [x] changelog is up to date

### CODE REVIEW

File: semgrep-core/src/typing/LSP_client.ml [hunk 1]
```diff
@@ -21,26 +21,27 @@ open Types
...
 (*****************************************************************************)
 (* Helpers *)
 (*****************************************************************************)
 
 let server =
-  match 1 with
-  | 1 -> "/home/pad/.opam/4.09.1/bin/ocamllsp"
+  match 0 with
+  | 0 -> "ocamllsp"
+  | 1 -> "/home/pad/.opam/4.12.0/bin/ocamllsp"
```
Comment (IagoAbal) [inline semgrep-core/src/typing/LSP_client.ml:44]: We should use `Sys.getenv "HOME"` here.

Comment (IagoAbal) [reply semgrep-core/src/typing/LSP_client.ml:44]: Ideally we should search for the binary in the `$PATH`.

**Comment by IagoAbal** [comment]
Very cool!

File: semgrep-core/src/typing/LSP_client.ml (hunk 1 above)
Comment (aryx) [reply semgrep-core/src/typing/LSP_client.ml:44]: yes, this is hardcoded for now because it's very experimental.

**Comment by aryx** [comment]
It's cool, but it does not really work :( It can infer local types, but I am not able to get
full type information for entities defined in other files. For example on many files where
ocamlmerlin says it's an AST_generic.expr under the cursor, ocamllsp just says 'a.

**Pull request merged**